File: platform/lib-java/src/main/java/com/canary/platform/lib/observability/CanaryLaneTag.java
package com.canary.platform.lib.observability;

import com.canary.platform.lib.XCanaryContext;

public final class CanaryLaneTag {

    public static final String STABLE = "stable";
    public static final String CANARY = "canary";

    private CanaryLaneTag() {}

    /** Returns the current lane tag value derived from {@link XCanaryContext}. */
    public static String current() {
        return XCanaryContext.isCanary() ? CANARY : STABLE;
    }
}

File: platform/lib-node/src/observability/canary-lane-tag.ts
import { isCanary } from "../x-canary-context.js";

export type Lane = "stable" | "canary";

export const STABLE: Lane = "stable";
export const CANARY: Lane = "canary";

/** Returns the current lane derived from XCanaryContext. */
export function currentLane(): Lane {
  return isCanary() ? CANARY : STABLE;
}
